src/main/resources/application-dev.properties

```diff
@@ -0,0 +1,12 @@
+# @ Author : Bijan Zohouri
+# 4/24/2018
+# This is Dev Env profile
+
+spring.datasource.url = jdbc:h2:mem:testdb
+spring.datasource.driverClassName = org.h2.Driver
+spring.datasource.username = sa
+spring.datasource.password =
+spring.jpa.show-sql = true
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
```

Comment:
Why do we need hibernate here ?

Reply:
71
down vote
accepted

First of all Dialect means "the variant of a language". Hibernate, as we know, is database agnostic. It can work with different databases. However, databases have proprietary extensions/native SQL variations, and set/sub-set of SQL standard implementations. Therefore at some point hibernate has to use database specific SQL. Hibernate uses "dialect" configuration to know which database you are using so that it can switch to the database specific SQL generator code wherever/whenever necessary.
We could have left this as default also.

Reply:
https://stackoverflow.com/questions/21012799/why-do-i-need-to-configure-the-sql-dialect-of-a-data-source